pyrene/src/components/DateTimePicker/RangeSelector/RangeSelector.tsx

```diff
@@ -0,0 +1,188 @@
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import clsx from 'clsx';
+import Button from '../../Button/Button';
+import {
+  getFutureDate, customDateFormat,
+  convertToUTCtime, convertToZoneTime, getClientTimeZone,
+} from '../../../utils/DateUtils';
+import TimeRangeSelector from '../../TimeRangeSelector/TimeRangeSelector';
+
+import styles from './RangeSelector.css';
+import DateTimePicker from '../DateTimePicker';
+import DetectClickOutside from './DetectClickOutside';
+
+type OnFunction = (value?: [number, number] | null) => void;
+export interface RangeSelectorProps{
+  /**
+   * Date format used by component
+   */
+  dateFormat?: string,
+  /**
+   * Boolean to toggle time display
+   */
+  dateOnly?: boolean,
+  /**
+  * This is a string array that represents the start and end labels of the component
+  */
+  labels?: [string, string]
+  /**
+   * This is a timestamp that represents the maximum date allowed by the component
+   */
+  maxDateTime?: number,
+  /**
+   * This is a timestamp that represents the minimum date allowed by the component
+   */
+  minDateTime?: number,
+  /**
+   * Function to handle onBlur event
+   */
+  onBlur?: OnFunction,
+  /**
+  * Function to handle onChange event
+  */
+  onChange: OnFunction,
+  /**
+   * Date format used by component
+   */
+  timeFormat?: string,
+  /**
+   * This is a unix timestamp, which is the number of seconds that have elapsed since Unix epoch
+   */
+  timeStamps?: [number, number],
+  /**
+   * This is must be a IANA time zone string
+   */
+  timeZone?: string,
+}
+
+/**
+ * A component for selecting a date and time range.
+ */
+const RangeSelector: React.FC<RangeSelectorProps> = (({
+  dateFormat = 'dd.MM.yyyy',
+  dateOnly = false,
+  maxDateTime = getFutureDate({ years: 1 }),
+  minDateTime = 0,
+  labels = ['From', 'To'],
+  onChange,
+  timeFormat = ' HH:mm',
+  timeStamps,
+  timeZone,
+}:RangeSelectorProps) => {
+
+  const [endDate, setEndDate] = useState<Date | undefined>();
+  const [startDate, setStartDate] = useState<Date | undefined>();
+
+  const [isCalOpen, setCalOpen] = useState<boolean | undefined>();
+  const [internaltTz, setTimezone] = useState('');
+
+  const clearValues = () => {
+    setStartDate(undefined);
+    setEndDate(undefined);
+  };
+
+  const handleApplyButton = (event: React.MouseEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (startDate && endDate) {
+      onChange([convertToUTCtime(startDate, internaltTz).valueOf(), convertToUTCtime(endDate, internaltTz).valueOf()]);
+    } else {
+      onChange(null);
+    }
+    setCalOpen(false);
+  };
+
+  const handleCancelButton = (event: React.MouseEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    clearValues();
+    onChange(null);
+    setCalOpen(false);
+  };
+
+  useEffect(() => {
+    setTimezone(timeZone || getClientTimeZone());
+  }, [timeZone]);
```

Comment:
I think this `useEffect` isn't necessary.

Reply:
What was agreed on was that the component will accept a time zone prop and if a time zone is not passed to the component, we get it from the user's browser.

Reply:
The `timeZone` props is not something that will be updated based on user input. Not super important anyhow.

Reply:
✅